Webapp/sources/services/data_export/eln_file.py

```diff
@@ -0,0 +1,555 @@
+                new_name = f"{original_name}_{count}{ext}"
+            file["display_name"] = new_name
+            file_names.append(new_name)
+
+    def get_parts(self):
+        """Define all parts for a reaction including the comments, files, and dataset"""
+        self._define_comments()
+        self._define_files()
+        self._define_reaction_dataset()
+
+    def _define_reaction_dataset(self):
+        """One reaction is its own dataset within the ELN file export and is defined here."""
+        self.defined_dataset = {
+            "@id": f"./{self.reaction.reaction_id}",
+            "@type": "Dataset",
+            "name": self.reaction.name,  # experiment title
+            "author": {"@id": f"./author/{self.reaction.creator_person.id}"},
+            "dateCreated": self.reaction.time_of_creation.strftime("%Y-%m-%d %H:%M:%S"),
+            "dateModified": self.reaction.time_of_update.strftime("%Y-%m-%d %H:%M:%S"),
+            "comment": [comment["@id"] for comment in self.defined_comments],
+            "hasPart": [{"@id": file["@id"]} for file in self.defined_files],
+            "description": self.reaction.description,
+            "text": self._get_summary_html(),  # todo restore after debugging
+            "encodingFormat": "text/html",
+            "url": url_for("main.index", _external=True)
+            + quote(
+                f"/{self.reaction.workbook.WorkGroup.name}/{self.reaction.workbook.name}/{self.reaction.reaction_id}/no"
+            ),
+            # may remove no in future
+        }
+
+    def _define_comments(self):
+        """Gets all the comments associated with a reaction and defines them in a format for the metadata json"""
+        for comment in self.reaction.addenda:
+            self.defined_comments.append(
+                {
+                    "@id": f"comment-id/{self.reaction.reaction_id}/{comment.id}",
+                    "@type": "Comment",
+                    "dateCreated": comment.time_of_creation,
+                    "text": comment.text,
+                    "author": {"@id": f"./author/{self.reaction.creator_person.id}"},
+                }
+            )
+
+    def _define_files(self):
+        """Gets all the file details associated with a reaction and defines them in a format for the metadata json"""
+        for file in self.files:
+            self.defined_files.append(
+                {
+                    "@id": f"./{self.reaction.reaction_id}/{file['display_name']}",
+                    "@type": "File",
+                    "name": f"{file['display_name']}",
+                    "description": self._get_file_description(file),
+                    "encodingFormat": file["mimetype"],
+                    "contentSize": str(file["content_size"]),
+                    "sha256": file["sha256"],
+                    "dateModified": file["time_of_creation"],
+                }
+            )
+
+    @staticmethod
+    def _get_file_description(file: Dict) -> str:
```

Comment:
Again here, if this method is only used inside this class you can remove the `@staticmethod` and add `self` as the first argument of the method

Reply:
See https://github.com/AI4Green/AI4Green/pull/69#discussion_r1666806707